fastestimator/dataset/batch_dataset.py

```diff
@@ -81,7 +81,18 @@ def _check_input(self) -> None:
         for num_sample in self.num_samples:
             assert isinstance(num_sample, int) and num_sample > 0, "only accept positive integer type as num_sample"
         # check dataset keys
-        dataset_keys = [set(dataset[0].keys()) for dataset in self.datasets]
+        dataset_keys = []
+        num_examples = self.num_samples * len(self.datasets) if len(
+            self.num_samples) == 1 else [x for x in self.num_samples]
+        for idx, dataset in enumerate(self.datasets):
+            sample_data = dataset[0]
```

Comment:
we should compute the fe_batch argument here since we're already incurring the cost of calling dataset[0]

Reply:
done